src/ape/managers/project.py

```diff
@@ -42,41 +86,128 @@ def sources(self) -> List[Path]:
+        contract_types = {
+            name: ct for name, ct in contract_types.items() if ct.sourceId not in deleted_sources
+        }
+
+        def file_needs_compiling(source: Path) -> bool:
+            source_path = str(source)
+            # New file added?
+            if source_path not in cached_sources:
+                return True
+
+            # File contents changed in source code folder?
+            checksum = compute_checksum(
+                source.read_text(),
+                algorithm=cached_sources[source_path].checksum.algorithm,
+            )
+            return checksum != cached_sources[source_path].checksum.hash
+
+        # NOTE: filter by checksum, etc., and compile what's needed
+        #       to bring our cached manifest up-to-date
+        needs_compiling = filter(file_needs_compiling, self.sources)
+        contract_types.update(self.compilers.compile(list(needs_compiling)))
+
+        # Update our cached contract types in our cached manifest
+        if manifest.contractTypes:
+            manifest.contractTypes.update(contract_types)
+        else:
+            manifest.contractTypes = contract_types
+
+        # Update our cached source code entries in our cached manifest
+        cached_sources = {
+            str(source): Source(  # type: ignore
+                checksum=Checksum(algorithm="md5", hash=compute_checksum(source.read_text())),
+                urls=[],
+            )
+            for source in self.sources
+        }
+        if manifest.sources:
+            manifest.sources.update(cached_sources)
+        else:
+            manifest.sources = cached_sources
+
+        # NOTE: Cache the updated manifest to disk (so `self.cached_manifest` reads next time)
+        self.manifest_cachefile.write_text(json.dumps(manifest.to_dict()))
+
+        return contract_types
+
     @property
     def contracts(self) -> Dict[str, ContractType]:
-        return self.compilers.compile(self.sources)
+        return self._load_contracts()
+
+    def __getattr__(self, attr_name: str):
+        contracts = self._load_contracts()
+        if attr_name in contracts:
+            return contracts[attr_name]
+        elif attr_name in self.dependencies:
+            return self.dependencies[attr_name]
+        else:
+            raise AttributeError(f"{self.__class__.__name__} has no attribute '{attr_name}'")
 
+    @property
     def _interfaces_folder(self) -> Path:
         return self.path / "interfaces"
 
+    @property
     def _scripts_folder(self) -> Path:
         return self.path / "scripts"
 
+    @property
     def _tests_folder(self) -> Path:
         return self.path / "tests"
 
     # TODO: Make this work for generating and caching the manifest file
+    @property
+    def compiler_data(self) -> List[Compiler]:
+        compilers = []
+
+        for compiler in self.compilers.registered_compilers.values():
+            for version in compiler.versions:
```

Comment:
To match up with change above (turn it into not a `property`)
```suggestion
        for extension, compiler in self.compilers.registered_compilers.items():
            for version in compiler.versions(file for file in self.sources if file.suffix == extension):
```

Reply:
Ok, it works:

```
$ ape console
In [1]: project.compiler_data
Out[1]: [Compiler(name='vyper', version='0.2.11', settings=None, contractTypes=None)]
```